src/server/common_services/auth_server.js

```diff
@@ -591,10 +591,6 @@ function _prepare_auth_request(req) {
         dbg.log3('load auth system:', req.system && req.system._id);
     };
 
-    req.has_bucket_permission = function(bucket) {
```

Comment:
has_s3_bucket_permission uses this function - not sure you can just remove. Also not sure what is our plan about this once allowed buckets are removed? to deny all requests or to allow them.

Reply:
@jackyalbo I'm fine with splitting this. @nimrod-becker Your thoughts?
As far as I understand, there is a different mechanism which is used instead of the "allowed buckets" one, so it's not a matter of allowing/denying everything as the other mechanism will take care of it (Nimrod - If I'm mistaken, please correct me).

Reply:
Regarding splitting, we can do that that is ok.

Allow/Deny will be be set according to the bucket policy, I believe, but might be mistaken, that Jacky is asking what would be the default in case there is no explicit rule. We should behave as S3 would, can easily test it on an S3 bucket

Reply:
Hey, I think that the default when there is no bucket policy should be to allow only the owner of the bucket, like AWS does ->

https://docs.aws.amazon.com/AmazonS3/latest/userguide/access-control-overview.html#about-resource-owner

>Amazon S3 bucket and object ownership
>
> Buckets and objects are Amazon S3 resources. By default, only the resource owner can access these resources. The resource owner refers to the AWS account that creates the resource. For example:
> 
> 1. The AWS account that you use to create buckets and upload objects owns those resources.
> 
> 2. If you upload an object using AWS Identity and Access Management (IAM) user or role credentials, the AWS account that the user or role belongs to owns the object.
> 
> 3. A bucket owner can grant cross-account permissions to another AWS account (or users in another account) to upload objects. In this case, the AWS account that uploads objects owns those objects. The bucket owner does not have permissions on the objects that other accounts own, with the following exceptions:
> 
>    - The bucket owner pays the bills. The bucket owner can deny access to any objects, or delete any objects in the bucket, regardless of who owns them.
> 
>     - The bucket owner can archive any objects or restore archived objects regardless of who owns them. Archival refers to the storage class used to store the objects. For more information, see Managing your storage lifecycle.
>